### ISSUE TITLE
problems in cross validation training

### ISSUE THREAD

**Issue by gina-belmonte** [ellisdg/3DUnetCNN#314]
When utilizing cross-validation, the 'train.py' script only executes the initial fold. The issue appears to stem from the fact that the script checks the log file to resume interrupted training, but starting from the second fold, it encounters epochs already computed. To address this for now, I resolved the problem by altering the training log file name for each fold. This was achieved by incorporating the following steps into the 'train.py' script before invoking 'run_training':

 confname = os.path.basename(config_filename).split(".")[0]    #Extract the base name of the configuration file
 trsep=training_log_filename.split(".") 
 trname = trsep[0]+"-"+ confname +"."+ trsep[1] #Create a new training log filename incorporating the configuration name
 
and then

run_training with **training_log_filename=trname**.

------------------------------------------------------------------------------------------

I also notice that the train script calculate the segmentation for validation and test sets but don't apply the activation function.

-------------------------------------------------------------------------------------------

I also report a minor problem in evaluate.py: 

in main()

labels = config["dataset"]["labels"]

instead of

labels = config["labels"]

-----------------------------------------------------------------------------------------------

Regards,
Gina

**Comment by ellisdg** [comment]
Hi @gina-belmonte

> When utilizing cross-validation, the 'train.py' script only executes the initial fold.

Thank you for letting me know. I will look into this today.

> I also notice that the train script calculate the segmentation for validation and test sets but don't apply the activation function.

The sigmoid activation function gets called by the loss function. That's why "sigmoid" is set to True in the configuration file for the Brats example.

> I also report a minor problem in evaluate.py:
> 
> in main()
> 
> labels = config["dataset"]["labels"]
> 
> instead of
> 
> labels = config["labels"]

Were you able to get the evaluate.py script to work at all? I need to go through and delete the scripts and files that aren't setup to work with the latest version of the project. I have made a lot of changes lately that make the code much faster and simpler, but it has broken many of the scripts.

**Comment by gina-belmonte** [comment]
Hi @ellisdg,

> > I also notice that the train script calculate the segmentation for validation and test sets but don't apply the activation function.
> 
> The sigmoid activation function gets called by the loss function. That's why "sigmoid" is set to True in the configuration file for the Brats example.

It doesn't seem to work for me, I will investigate further.
 
> Were you able to get the evaluate.py script to work at all? I need to go through and delete the scripts and files that aren't setup to work with the latest version of the project. I have made a lot of changes lately that make the code much faster and simpler, but it has broken many of the scripts.

Actually, I wasn't able to utilize the script because I'm uncertain about how to invoke it. I thought it might be able to compute the loss (Dice?) by comparing a test set with its corresponding ground truth, but I'm not entirely sure of the procedure.

**Comment by ellisdg** [comment]
@gina-belmonte, are you setting the training log filename on the commandline or using the default?

**Comment by gina-belmonte** [comment]
Hi,
thank you.

Regarding the activation function, I have set "sigmoid": true in the config in the loss section and specified the "test_filenames". This way, the training script calculates predictions for each fold both on the validation and the test set. However, in both cases, the predictions are not segmentations but rather unnormalized maps. On the other hand, when I use the predict script with the "--activation sigmoid" option, the output consists of segmentations.

**Comment by ellisdg** [comment]
I see. I ran into this issue too. I'll see if I can fix it soon.

**Comment by ellisdg** [comment]
The issue should be fixed now. Reopen this issue if it persists. Thanks!
